### PR TITLE
Binance: replace deprecated /v1 API calls

### PR DESCRIPTION
**Pull request by dackroyd**
# PR Description

Several /v1 API calls have been deprecated, and will be removed shortly. Moving these across to their /v3 equivalents

Capturing of the klines request via the recorder was failing, as it did not handle the structure where the klines response in an array of mixed primitive arrays. Excluding primitive values from the exclusion checks addresses this

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Travis with my changes

### CODE REVIEW

**Comment by dackroyd** [comment]
https://binance-docs.github.io/apidocs/spot/en/#change-log

> Deprecation of v1 endpoints:
>
> By end of Q1 2020, the following endpoints will be removed from the API. The documentation has been updated to use the v3 versions of these endpoints.
>
> * GET api/v1/depth
> * GET api/v1/historicalTrades
> * GET api/v1/aggTrades
> * GET api/v1/klines
> * GET api/v1/ticker/24hr
> * GET api/v1/ticker/price
> * GET api/v1/exchangeInfo
> * POST api/v1/userDataStream
> * PUT api/v1/userDataStream
> * GET api/v1/ping
> * GET api/v1/time
> * GET api/v1/ticker/bookTicker
>
> **These endpoints however, will NOT be migrated to v3. Please use the following endpoints instead moving forward.**
>
> Old V1 Endpoints | New V3 Endpoints
> -- | --
> GET api/v1/ticker/allPrices | GET api/v3/ticker/price
> GET api/v1/ticker/allBookTickers | GET api/v3/ticker/bookTicker

File: exchanges/binance/binance_types.go [hunk 1]
```diff
@@ -247,7 +245,7 @@ type PriceChangeStats struct {
 	QuoteVolume        float64 `json:"quoteVolume,string"`
 	OpenTime           int64   `json:"openTime"`
 	CloseTime          int64   `json:"closeTime"`
-	FirstID            int64   `json:"fristId"`
+	FirstID            int64   `json:"firstId"`
```
Comment (shazbert) [inline exchanges/binance/binance_types.go:248]: nice catch!

File: exchanges/binance/binance.go [hunk 2]
```diff
@@ -26,14 +26,14 @@ const (
...
-	exchangeInfo      = "/api/v1/exchangeInfo"
-	orderBookDepth    = "/api/v1/depth"
-	recentTrades      = "/api/v1/trades"
-	historicalTrades  = "/api/v1/historicalTrades"
-	aggregatedTrades  = "/api/v1/aggTrades"
-	candleStick       = "/api/v1/klines"
+	exchangeInfo      = "/api/v3/exchangeInfo"
+	orderBookDepth    = "/api/v3/depth"
+	recentTrades      = "/api/v3/trades"
+	historicalTrades  = "/api/v3/historicalTrades"
```
Comment (shazbert) [inline exchanges/binance/binance.go:32]: We haven't gotten functionality for this, will make a note to expand and use the v3 endpoint. Unless you want to add it while you are here? If not all good let me know.

File: exchanges/mock/recording.go [hunk 3]
```diff
@@ -288,12 +288,20 @@ func CheckJSON(data interface{}, excluded *Exclusion) (interface{}, error) {
...
+			switch v.(type) {
+			case map[string]interface{}, []interface{}:
+				checkedData, err := CheckJSON(v, excluded)
+				if err != nil {
+					return nil, err
+				}
 
-			sData = append(sData, checkedData)
+				sData = append(sData, checkedData)
+				continue
```
Comment (shazbert) [inline exchanges/mock/recording.go:300]: I think we can remove this continue.

Comment (dackroyd) [reply exchanges/mock/recording.go:300]: Whoops! Ta, this was from an earlier revision of the change where the switch didn't have a default, and instead happened after. I'll remove it now

File: exchanges/binance/binance.go (hunk 2 above)
Comment (dackroyd) [reply exchanges/binance/binance.go:32]: Will keep that for a follow up change I think 👌

Comment (shazbert) [reply exchanges/binance/binance.go:32]: Perfect 👍

**Pull request merged**